dissect/target/loaders/acquire.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import logging
+import zipfile
+from pathlib import Path
+
+from dissect.target.filesystems.tar import TarFilesystem
+from dissect.target.filesystems.zip import ZipFilesystem
+from dissect.target.loader import Loader
+from dissect.target.loaders.dir import find_and_map_dirs
+from dissect.target.target import Target
+
+log = logging.getLogger(__name__)
+
+FILESYSTEMS_ROOT = "fs"
+FILESYSTEMS_LEGACY_ROOT = "sysvol"
+
+
+def _get_root(path: Path):
+    if path.is_file():
+        fh = path.open("rb")
+        if TarFilesystem._detect(fh):
+            return TarFilesystem(fh).path()
+
+        # test this
+        if ZipFilesystem._detect(fh):
+            return zipfile.Path(path.open("rb"))
+
+    return None
+
+
+class AcquireLoader(Loader):
+    def __init__(self, path: Path, **kwargs):
+        super().__init__(path)
+
+        self.root = _get_root(path)
+
+    @staticmethod
+    def detect(path: Path) -> bool:
+        root = _get_root(path)
+
+        if not root:
+            return False
+
+        return root.joinpath(FILESYSTEMS_ROOT).exists() or root.joinpath(FILESYSTEMS_LEGACY_ROOT).exists()
```

Comment:
This will be very slow on large tar files, where there will be a double performance penalty. First the entire tar file will be parsed just for this check, then if it exists, it will be parsed again in `__init__`. However, if it doesn't exist, it will still be parsed _again_ in the actual tar loader.

Reply:
Yes I agree, but is there a way to prevent that? As we do need to check if the `fs` directory exists in the file, since otherwise the tar is not an acquire collect and thus need to be handled by the `TarLoader`.

Reply:
"Ideally" (not really, since it's only for performance and the code path itself will be more confusing) it's a subpath in the `zip` and `tar` loaders. So you piggy back on the detection logic of those, and upon mapping you do a fast check to see if you need to divert logic to Acquire logic (which could exist in `loaders/acquire.py`.

At least, that's the "best" idea I can come up with right now.

Reply:
Can you base it on https://github.com/fox-it/dissect.target/pull/1000?